numpy/core/tests/test_multiarray.py

```diff
@@ -2329,14 +2329,14 @@ def test_dot_type_mismatch(self):
     def test_dot_out_mem_overlap(self):
         np.random.seed(1)
 
-        for dtype in [np.object_, np.float32, np.complex128, np.int64]:
-            a0 = np.random.rand(3, 3).astype(dtype)
-            b0 = np.random.rand(3, 3).astype(dtype)
-            for a, b in [(a0.copy(), b0.copy()),
-                         (a0.copy().T, b0.copy())]:
-                y = np.dot(a, b)
-                x = np.dot(a, b, out=b)
-                assert_equal(x, y, err_msg=repr(dtype))
+        # Test BLAS and non-BLAS code paths
+        for dtype in [np.float32, np.int64]:
```

Comment:
Actually, I prefer the maximal tests because the code may change in the future. The tests not only check current behavior, they also define correct behavior for the future.